core/Container/ContainerFactory.php

```diff
@@ -82,4 +82,16 @@ private function addEnvironmentConfig(ContainerBuilder $builder)
 
         $builder->addDefinitions($file);
     }
+
+    private function addPluginConfigs(ContainerBuilder $builder)
+    {
+        $plugins = Config::getInstance()->Plugins['Plugins'];
+
+        foreach ($plugins as $plugin) {
+            $file = Manager::getPluginsDirectory() . $plugin . '/plugin.php';
+            if (file_exists($file)) {
```

Comment:
I've implemented `plugins/MyPlugin/config.php`, see the diff. Not the most efficient piece of code ever but I guess the opcache should make it a non-problem.

I like having `config.php` at the root of the plugin. But if the change is really not efficient, we could move it to a sub-directory.

But for 3.0 we could say `config.php` must be a config file, move your class elsewhere. In the meantime we keep BC. So the current `Config` classes would be sort of "deprecated".